internal/cloud/backend_plan.go

```diff
@@ -362,6 +369,39 @@ in order to capture the filesystem context the remote workspace expects:
 	return r, nil
 }
 
+// AssertImportCompatible errors if the user is attempting to use configuration-
+// driven import and the version of the agent or API is too low to support it.
+func (b *Cloud) AssertImportCompatible(config *configs.Config) error {
+	// Check TFC_RUN_ID is populated, indicating we are running in a remote TFC
+	// execution environment.
+	if len(config.Module.Import) > 0 && os.Getenv("TFC_RUN_ID") != "" {
+		// First, check the remote API version is high enough.
+		currentAPIVersion, err := version.NewVersion(b.client.RemoteAPIVersion())
+		if err != nil {
+			return fmt.Errorf("Error parsing remote API version. To proceed, please remove any import blocks from your config. Please report the following error to the Terraform team: %s", err)
+		}
+		desiredAPIVersion, _ := version.NewVersion("2.6")
+		if currentAPIVersion.LessThan(desiredAPIVersion) {
+			return fmt.Errorf("Import blocks not available in this version of Terraform. You are using API version %s, but this feature requires version %s. Please remove any import blocks from your config or upgrade TFC.", currentAPIVersion, desiredAPIVersion)
+		}
+
+		// Second, check the agent version is high enough.
+		agentEnv, isSet := os.LookupEnv("TFC_AGENT_VERSION")
+		if !isSet {
+			return fmt.Errorf("Error reading TFC agent version. To proceed, please remove any import blocks from your config. Please report the following error to the Terraform team: TFC_AGENT_VERSION not present.")
```

Comment:
Is this correct? I am assuming based on the spec that if all other conditions are true but `TFC_AGENT_VERSION` is not set at all, then the user is using an old version of the agent.

On the other hand, it could be that it is valid for this to be unset (perhaps you're not using the agent at all?) and it should fall through.

cc @radditude

Reply:
That's correct! If someone was using the `cloud` backend in local execution mode where TFC is used only for state storage, `TFC_RUN_ID` wouldn't be set (unless they've set it intentionally for some reason, in which case I would like to understand their use case).  `TFC_AGENT_VERSION` is a recent addition, so I'd expect that it being unset will be a common case.